xapian-core/include/xapian/weight.h

```diff
@@ -526,15 +526,20 @@ class XAPIAN_VISIBILITY_DEFAULT BM25Weight : public Weight {
      *		   document length normalisation of wdf is.  0 means no
      *		   normalisation; 1 means full normalisation.  (default 0.5)
      *
+     *  @param delta  A parameter for pseudo tf value to control the scale
+     * 		   of the tf lower bound. Delta(δ) can be tuned for e.g from 0.0 
+     * 		   to 1.5 but BM25+ can still work effectively across collections 
+     *		   with a fixed δ = 1.0. (default 1.0)
+     * 
      *  @param min_normlen  A parameter specifying a minimum value for
      *		   normalised document length.  Normalised document length
      *		   values less than this will be clamped to this value, helping
      *		   to prevent very short documents getting large weights.
      *		   (default 0.5)
      */
-    BM25Weight(double k1, double k2, double k3, double b, double min_normlen)
+    BM25Weight(double k1, double k2, double k3, double b, double delta, double min_normlen)
```

Comment:
You mean an overloaded constructor with `param_delta` initialized to `1.0`and rest of the parameters initialized to values received in variables k1, k2 etc.? 
So, initializing `param_delta` to `1.0` seem to be overriding `delta` value with `1.0` which calls additional stats required for BM25+ but we can avoid calling stats, `WQF`, `DOC_LENGTH` and `AVERAGE_LENGTH`, when `delta` is zero or less than zero. 
In that case, we'll have a new constructor looking like :-
```C++
BM25Weight(double k1, double k2, double k3, double b, double min_normlen, double delta)
    : param_k1(k1), param_k2(k2), param_k3(k3), param_b(b),
      param_min_normlen(min_normlen), param_delta(delta)
    {
        param_delta = 1.0;
        need_stat(WQF);
	    need_stat(DOC_LENGTH);
	    need_stat(AVERAGE_LENGTH);
	    if (param_k1 < 0) param_k1 = 0;
	    if (param_k2 < 0) param_k2 = 0;
	    if (param_k3 < 0) param_k3 = 0;
	    if (param_b < 0) {
	        param_b = 0;
	        } else if (param_b > 1) {
	        param_b = 1;
	    }
	    need_stat(COLLECTION_SIZE);
	    need_stat(RSET_SIZE);
	    need_stat(TERMFREQ);
	    need_stat(RELTERMFREQ);
	    need_stat(WDF);
	    need_stat(WDF_MAX);
	    if (param_k2 != 0 || (param_k1 != 0 && param_b != 0)) {
	        need_stat(DOC_LENGTH_MIN);
	        need_stat(AVERAGE_LENGTH);
	    }
	    if (param_k1 != 0 && param_b != 0) need_stat(DOC_LENGTH);
	    if (param_k2 != 0) need_stat(QUERY_LENGTH);
	    if (param_k3 != 0) need_stat(WQF);
    }
```
Does this look good to you?